src/neo/SmartContract/Native/NeoToken.cs

```diff
@@ -332,20 +333,55 @@ private async ContractTask<bool> Vote(ApplicationEngine engine, UInt160 account,
         }
 
         /// <summary>
-        /// Gets all registered candidates.
+        /// Gets the first 256 registered candidates.
         /// </summary>
         /// <param name="snapshot">The snapshot used to read data.</param>
         /// <returns>All the registered candidates.</returns>
         [ContractMethod(CpuFee = 1 << 22, RequiredCallFlags = CallFlags.ReadStates)]
-        public (ECPoint PublicKey, BigInteger Votes)[] GetCandidates(DataCache snapshot)
+        private (ECPoint PublicKey, BigInteger Votes)[] GetCandidates(DataCache snapshot)
+        {
+            return GetCandidatesInternal(snapshot)
+                .Select(p => (p.PublicKey, p.State.Votes))
+                .Take(256)
+                .ToArray();
+        }
+
+        /// <summary>
+        /// Gets the registered candidates iterator.
+        /// </summary>
+        /// <param name="engine">The <see cref="ApplicationEngine"/> that is executing the contract.</param>
+        /// <returns>All the registered candidates.</returns>
+        [ContractMethod(CpuFee = 1 << 22, RequiredCallFlags = CallFlags.ReadStates)]
+        private IIterator GetAllCandidates(ApplicationEngine engine)
```

Comment:
I'm glad you've added this, but we still have no mechanism to retrieve all the values of an iterator over JSON-RPC